resources/persist/src/lib.rs

```diff
@@ -41,6 +47,41 @@ impl PersistInstance {
         Ok(serialize_into(&mut writer, &struc).map_err(PersistError::Serialize))?
     }
 
+    /// list method returns a vector of strings containing all the keys associated with a PersistInstance
+    pub fn list(&self) -> Result<Vec<String>, PersistError> {
+        let storage_folder = self.get_storage_folder();
+
+        let mut list = Vec::new();
+
+        let entries = fs::read_dir(storage_folder).map_err(PersistError::ListFolder)?;
+        for entry in entries {
+            let dir = entry.map_err(PersistError::ListFolder)?;
+            list.push(
+                dir.path()
+                    .into_os_string()
+                    .into_string()
+                    .unwrap_or_else(|_| "path contains non-UTF-8 characters".to_string())
+                    .split(".bin")
+                    .collect::<String>(),
+            );
+        }
+        Ok(list)
+    }
+
+    /// clear method removes the storage folder from the PersistInstance
+    pub fn clear(&self) -> Result<(), PersistError> {
+        let folder_path = self.get_storage_folder();
+        fs::remove_dir_all(folder_path).map_err(PersistError::ClearFolder)?;
+        Ok(())
+    }
+
+    /// remove method deletes a key from the PersistInstance
+    pub fn remove(&self, key: String) -> Result<(), PersistError> {
```

Comment:
Since save and load takes the key as a `&str`, it makes sense that this also does. This means that you will need `.as_str()` in the test case.

Reply:
Thanks.  Waffled endlessly here.  Thanks for confirming that staying consistent with the rest is best.

Reply:
Now using .as_str() in the assocated test case.